pennylane/decomposition/__init__.py

```diff
@@ -0,0 +1,148 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""
+.. currentmodule:: pennylane
+
+This module implements the infrastructure for PennyLane's new graph-based decomposition system.
+
+.. warning::
+
+    This module is experimental and is subject to change in the future.
+
+.. currentmodule:: pennylane.decomposition
+
+.. autosummary::
+    :toctree: api
+
+    ~register_resources
+    ~resource_rep
+    ~controlled_resource_rep
+    ~adjoint_resource_rep
+    ~add_decomps
+    ~list_decomps
+    ~has_decomp
+    ~Resources
+    ~CompressedResourceOp
+    ~DecompositionRule
+
+.. TODO::
+    Add section here explaining how to enable and disable the new decomposition system. [sc-83993]
```

Comment:
```suggestion
To activate and deactivate the new PennyLane program capturing mechanism, use the switches :func:`~.pennylane.capture.enable` and :func:`~.pennylane.capture.disable_graph`. 
```

Do we have something we can use to query if the new system is available or not? Similar to `qml.capture.enabled/disabled`.

Reply:
This PR doesn't contain this. Everything is this PR is going to be available regardless of whether the new decompositions system is enabled or not. The toggle would only affect how the `decompose` transform behaves.